VAMobile/src/api/queryClient.ts

```diff
@@ -0,0 +1,20 @@
+import { QueryCache, QueryClient } from '@tanstack/react-query'
+import { UserAnalytic, logNonFatalErrorToFirebase, setAnalyticsUserProperty } from 'utils/analytics'
+import { isErrorObject } from 'utils/common'
+
+export default new QueryClient({
+  queryCache: new QueryCache({
+    onSuccess: async (data, query) => {
+      const analyticsUserProperty = query?.meta?.analyticsUserProperty as UserAnalytic
+      if (analyticsUserProperty) {
+        await setAnalyticsUserProperty(analyticsUserProperty)
+      }
+    },
+    onError: (error, query) => {
+      const errorName = query?.meta?.errorName as string
+      if (errorName && isErrorObject(error)) {
+        logNonFatalErrorToFirebase(error, errorName)
+      }
+    },
+  }),
```

Comment:
The query cache can access data from the `meta` option in the `useQuery` hook. This allows us to reduce redundancy in queries by adding common function calls.

The `useQuery` hook also has `onSuccess` and `onError` options where we can call common post API call functions like `logNonFatalErrorToFirebase` and `setAnalyticsUserProperty` from, but those options are going to be removed from the hook in the next major release (v5), so using the query cache is the recommended alternative.

Here's an example of how to pass data through the `meta` option:
https://github.com/department-of-veterans-affairs/va-mobile-app/pull/6627/files#diff-e5f8dbfb8c5a6bd856e1ebd275a257052bc3503915bc6b4ead9530800a86d9cfR26